File: C#/1254 - Substituição de Tag.cs
/// author Josélio de S. C. Júnior <joseliojrx25@gmail.com>
/// copyright Josélio de S. C. Júnior 2021

using System;
using System.Linq;
using System.Text.RegularExpressions;

class URI {
  static void Main(string[] args) { 
    while (true) {
      string tag = Console.ReadLine() ?? "";
      if (tag == "") break;

      string  rv = Console.ReadLine() ?? "", code = Console.ReadLine() ?? "";
    
      foreach (Match e in Regex.Matches(code, @"<.*?>")) {
        string rp = Regex.Replace(e.ToString(), $"(?i){tag}", rv);
        code = Regex.Replace(code, e.ToString(), rp);
      }
    
      Console.WriteLine(code);
    }
  }
}

File: Python/1254 - Substituição de Tag.py
# author: Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
# copyright: Josélio Júnior (Lunatic Fox) 2022

import re
while True:
  try:
    r: list[str] = []
    for _ in range(3):
      r.append(input())
    tag, rep, ln = r

    r_tag = re.findall(fr'(?i)(<.*?>)', ln)
    rp_tag = list(map(lambda e: re.sub(fr'(?i){tag}', rep, e), r_tag))

    for i in range(len(r_tag)):
      ln = ln.replace(r_tag[i], rp_tag[i])

    print(ln)
  except EOFError:
    break
